syncd/SaiSwitch.cpp

```diff
@@ -948,46 +948,50 @@ void SaiSwitch::redisUpdatePortLaneMap(
 
 void SaiSwitch::onPostPortCreate(
         _In_ sai_object_id_t port_rid,
-        _In_ sai_object_id_t port_vid)
+        _In_ sai_object_id_t port_vid,
+        _In_ bool discoverPortObjects)
```

Comment:
i feel like this eintrie change in this function is overcomplicated, it sholud be something like this:

```C++

if (object_type(oid) == SAI_OBJECT_TYPE_PORT && shouldSkipPorts)
  continue;

Reply:
@kcudnik 
Regarding ```object_type(oid) == SAI_OBJECT_TYPE_PORT```, the function is called ```onPostPortCreate``` so unless someone is calling it on object other than port I don't think this check is needed.

Do you mean early return? Like:

```c++
redisUpdatePortLaneMap(port_rid);

if (!discoverPortObjects)
{
    return;
}

...
```

Reply:
ah, i thoung you also modify discover process, since it will also discover all objects on all ports, so i guess on cold boot you only need onpostportcreate, but this could still crash on next fast-boot

please do couple of fst-boot to fast-boot reboots with your patch to see if this will wrok

Reply:
looking at taht code, you only need to modify SaiDiscovery process with flag to ignore port discovery, no else code is needed to be changed

Reply:
and ig you look on master, in SaiDiscovery.cpp file at line 34, you can actually pass new flag - to not discover port objects over VendorSaiOptions class to not  forward all bool arguments to discover ports, if you want

Reply:
@kcudnik Thanks, looking into VendorSaiOptions, that is new in master. Will check whether I can utilize it to avoid parameter passing.

What I tested few month ago was 202311 with this patch:
cold boot -> fast boot -> fast boot
cold boot -> fast boot -> warm boot

Will double check on latest master.

Reply:
Just simple reboot maybe not enough, I'm maybe this required: cold boot, fast boot, query ingress priority groups and assign something like all or query port queues, ten warm boot and again warm boot etc, there must be operation that will query objects that were not queried in discovery

Reply:
And this dlag should be command line options like skip port discovery with just couple line changes I a pro o used in this comment

Reply:
> Just simple reboot maybe not enough, I'm maybe this required: cold boot, fast boot, query ingress priority groups and assign something like all or query port queues, ten warm boot and again warm boot etc, there must be operation that will query objects that were not queried in discovery

@kcudnik This is how the test was done. Orchagent queries PGs, queues that were not queried during discovery - https://github.com/sonic-net/sonic-swss/blob/master/orchagent/portsorch.cpp#L5738




> And this dlag should be command line options like skip port discovery with just couple line changes I a pro o used in this comment

The command line option means someone can disable it at runtime and get broken fast boot, we would like to avoid that.

Reply:
yes, since we don want to skip port ddetection on other platforms than yours

Reply:
and you can disable those ports in init script for your platform only

Reply:
@kcudnik Platform is known at compile time, syncd is compiled differently for different platforms. My change as is right now should not affect other platforms. This change purpose is to improve startup time, however with platform detection done in script I will add some additional CPU cycles for that. Even though it is very small, on some lower systems the init scripts execution time is worse, that's why I am leaning towards moving all to compile time if possible.

Reply:
if it's know at compile time, put this:

```c
#ifdef nvidia
if (object_type(oid) == SAI_OBJECT_TYPE_PORT)
  continue;
#endif
```
in si discovery

Reply:
also add  log warn message, that discovery port was disabled on nvidia platform

Reply:
@kcudnik Yes, but not on get operations, since this is not required for fast/warm boot for us. Priority is on set operations.